### PR TITLE
[SPARK-45819][CORE] Support `clear` in REST Submission API

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to support `clear` in REST Submission API to clear `completed` drivers and apps.

### Why are the changes needed?

This new feature is helpful for users to reset the completed drivers and apps in Spark Master.

**"1 Completed"**
Screenshot 2023-11-07 at 12 56 02 AM

**After invoking `clear` API, "0 Completed"**
```
$ curl -X POST http://max.local:6066/v1/submissions/clear
{
  "action" : "ClearResponse",
  "message" : "",
  "serverSparkVersion" : "4.0.0-SNAPSHOT",
  "success" : true
}
```

Screenshot 2023-11-07 at 12 56 24 AM

### Does this PR introduce _any_ user-facing change?

No, this is a new API.

### How was this patch tested?

Pass the CIs with the newly added test case.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this when you have some time, @viirya ?

File: core/src/test/scala/org/apache/spark/deploy/rest/StandaloneRestSubmitSuite.scala [hunk 1]
```diff
@@ -660,4 +682,15 @@ private class FaultyStandaloneRestServer(
       s
     }
   }
+
+  /** A faulty status servlet that explodes. */
```
Comment (viirya) [inline core/src/test/scala/org/apache/spark/deploy/rest/StandaloneRestSubmitSuite.scala:686]: nit:

```suggestion
  /** A faulty clear servlet that explodes. */
```

**Review by viirya — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/deploy/rest/StandaloneRestSubmitSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/deploy/rest/StandaloneRestSubmitSuite.scala:686]: Thank you!

**Comment by dongjoon-hyun** [comment]
Thank you for review and approval, @viirya . Merged to master for Apache Spark 4.0.0.

**Pull request closed without merging**